packages/aws-cdk/lib/context-providers/vpcs.ts

```diff
@@ -0,0 +1,193 @@
+import cxapi = require('@aws-cdk/cx-api');
+import AWS = require('aws-sdk');
+import { Mode, SDK } from '../api';
+import { debug } from '../logging';
+import { ContextProviderPlugin } from './provider';
+
+export class VpcNetworkContextProviderPlugin implements ContextProviderPlugin {
+
+  constructor(private readonly aws: SDK) {
+  }
+
+  public async getValue(args: cxapi.VpcContextQuery) {
+    const account: string = args.account;
+    const region: string = args.region;
+
+    const ec2 = await this.aws.ec2(account, region, Mode.ForReading);
+
+    const vpcId = await this.findVpc(ec2, args);
+
+    return await this.readVpcProps(ec2, vpcId);
+  }
+
+  private async findVpc(ec2: AWS.EC2, args: {[key: string]: any}): Promise<string> {
+    const vpcId: string | undefined = args.vpcId;
+    const vpcName: string | undefined = args.vpcName;
+    const tags: {[key: string]: string} | undefined = args.tags;
+    const isDefault: boolean | undefined = args.isDefault;
+
+    // Builter request filter
+    const filters: AWS.EC2.Filter[] = [];
+    if (vpcId) { filters.push({ Name: 'vpc-id', Values: [vpcId] }); }
```

Comment:
Why not just expose the filters API in the context provider API and move the logic of building the filters to the construct library?

Reply:
@rix0rrr might still be recovering from when I used the word `filter` and he really didn't like it, but this context is different and I won't give him *too* hard a time if `filter` comes back.

Reply:
I guess I wanted as narrow an API as possible. But yeah, I suppose it makes it easier to add features in the future.

Reply:
Just feels like we can expose this to the user and give them more power without sacrificing too much usability (maybe we can add a bunch of sugar methods for the common use cases)